test/pubsub-in-browser.spec.js

```diff
@@ -0,0 +1,179 @@
+        describe('.publish', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.publish(topic, 'hello friend', (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.peers(topic, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.ls', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.ls((err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+      })
+
+      describe('Promise API', () => {
+        describe('.publish', () => {
+          it('throws an error', () => {
+            return ipfs1.pubsub.publish(topic, 'hello friend')
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+              })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler)
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+                done()
+              })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.peers(topic)
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+                done()
+              })
+          })
+        })
+
+        describe('.ls', () => {
+          it('throws an error', () => {
+            return ipfs1.pubsub.ls()
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+              })
+          })
+        })
+      })
+
+      describe('.unsubscribe', () => {
+        it('throws an error', (done) => {
+          try {
+            ipfs1.pubsub.unsubscribe(topic)
+            done('unsubscribe() didn\'t throw an error')
+          } catch (e) {
+            expect(e).to.exist
+            expect(e.message).to.equal(expectedError)
+            done()
+          }
+        })
+      })
+    })
+  })
```

Comment:
In order to make sure we throw the error when used in the browser, I added these tests as separate ones from the interface ones. See the spec file descriptions and comments. These tests are only to test that we error correctly when js-ipfs-api is used in the browser (as per our decision to not support it for now). These shouldn't go to the interface as they would error js-ipfs, too (which is not what we want). Makes sense?